packages/react/src/components/Dialog/index.tsx

```diff
@@ -18,9 +18,10 @@ export interface DialogProps extends React.HTMLAttributes<HTMLDivElement> {
   onClose?: () => void;
   forceAction?: boolean;
   heading:
+    | string
```

Comment:
As a result of writing tests, I discovered this type was too restrictive so expanding the type to allow for strings to be set as well.